core/src/main/java/aima/core/search/basic/uninformed/BidirectionalSearch.java

```diff
@@ -22,68 +18,87 @@
  * frontiers.
  * @author manthan
  */
-public class BidirectionalSearch<A, S> implements SearchForActionsBidirectionallyFunction<A, S> {
+public class BidirectionalSearch<A, S> implements SearchForActionsBidirectionallyFunction<A, S>, BidirectionalActions<A> {
```

Comment:
Not a good idea having the search algorithm implement BidirectionalActions<A>. What if apply is called twice with two different arguments? Ideally, just create a simple:

aima.core.search.basic.support.BasicBidirectionalActions

class and use it.